test/test_checks.py

```diff
@@ -171,5 +171,61 @@ class Options():
         ])
 
 
+class AvoidEscapeTestChecks(TestCase):
```

Comment:
I feel like a lot of these are redundant and thus going to increase maintenance work =/ Can we consolidate them with existing tests instead? Maybe make the tests be `class SinglesAllowEscaped`/`SinglesAvoidEscaped`/etc

Reply:
Sure, I'll have a think about how best to do this :)